.github/workflows/publish-gh-release.yml

```diff
@@ -0,0 +1,53 @@
+name: Publish to Github
+
+on:
+  push:
+    tags:
+      - "v*.*.*"
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  release:
+    env:
+      GENERATED_CHANGELOG: /tmp/generated_changelog.txt
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Generate Changelog
+        run: |
+          python << EOF > ${{ env.GENERATED_CHANGELOG }}
+          import re
+
+          include  = False
+          lines = []
+
+          with open("${GITHUB_WORKSPACE}/changelog.txt") as f:
```

Comment:
I don't like using bash variable interpolation inside of the Python script. It wouldn't work as a standalone Python script, which I think is important here. Can we use `os.environ` instead? (Addressing this in my follow-up PR)